app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HallsController < ApplicationController
+
+    before_action :set_hall, only: [:show, :update, :destroy]
```

Comment:
I think that simply calling `set_hall` in any of these actions will be a better choice.
[Here](https://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html) you can read why.